File: element-common/src/main/java/io/github/solusmods/eternalcore/element/EternalCoreElements.java
package io.github.solusmods.eternalcore.element;

import io.github.solusmods.eternalcore.element.impl.ElementRegistry;
import io.github.solusmods.eternalcore.element.impl.ElementsStorage;
import io.github.solusmods.eternalcore.element.impl.network.ElementsNetwork;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraft.resources.ResourceLocation;

import static io.github.solusmods.eternalcore.element.ModuleConstants.MOD_ID;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class EternalCoreElements {

    public static ResourceLocation create(String path) {
        return ResourceLocation.fromNamespaceAndPath(MOD_ID, path);
    }

    public static void init(){
        ElementRegistry.init();
        ElementsStorage.init();
        ElementsNetwork.init();
    }
}

File: element-common/src/main/kotlin/io/github/solusmods/eternalcore/element/EternalCoreElements.kt
package io.github.solusmods.eternalcore.element

import io.github.solusmods.eternalcore.element.impl.ElementRegistry
import io.github.solusmods.eternalcore.element.impl.ElementsStorage
import io.github.solusmods.eternalcore.element.impl.network.ElementsNetwork
import net.minecraft.resources.ResourceLocation

object EternalCoreElements {
    fun create(path: String): ResourceLocation {
        return ResourceLocation.fromNamespaceAndPath(ModuleConstants.MOD_ID, path)
    }

    @JvmStatic
    fun init() {
        ElementRegistry.init()
        ElementsStorage.init()
        ElementsNetwork.init()
    }
}
